### PR TITLE
Fix Manually Linking Util

### PR DESCRIPTION
**Pull request by revam**

### CODE REVIEW

**Comment by revam** [comment]
The logic to save is still a bit buggy in regards to RTK's cache. Will look at it some more tomorrow maybe, unless any of you know how to fix this.

File: src/pages/utilities/UnrecognizedUtilityTabs/LinkFilesTab.tsx [hunk 1]
```diff
@@ -232,33 +242,108 @@ function LinkFilesTab() {
...
-      forEach(fileIds, (link) => {
-        if (link.FileID === 0) {
-          return;
+  useEffect(() => {
+    if (!isLinking || isLinkingRunning || getAnidbSeriesQuery.isLoading || !(getAnidbSeriesQuery.isSuccess && getAnidbSeriesQuery.data.ShokoID)) {
+      return;
+    }
+
+    const seriesID = getAnidbSeriesQuery.data.ShokoID;
+    const linkFiles = async () => {
```
Comment (hidden4003) [inline src/pages/utilities/UnrecognizedUtilityTabs/LinkFilesTab.tsx:251]: seems to me that linkFiles does not have to be inside the useEffect it only makes it look a lot more complicated than it should be

File: src/pages/utilities/UnrecognizedUtilityTabs/LinkFilesTab.tsx [hunk 2]
```diff
@@ -232,33 +242,108 @@ function LinkFilesTab() {
...
+      if (isLinkingRunning) return;
```
Comment (hidden4003) [inline src/pages/utilities/UnrecognizedUtilityTabs/LinkFilesTab.tsx:252]: This is duplicate of top condition

Comment (hidden4003) [reply src/pages/utilities/UnrecognizedUtilityTabs/LinkFilesTab.tsx:252]: Some explanation on what feature is being added here would be nice.

Comment (revam) [reply src/pages/utilities/UnrecognizedUtilityTabs/LinkFilesTab.tsx]: > Some explanation on what feature is being added here would be nice.

We're making sure the Shoko series is only created when the 'Save' button is pressed, and not before that.

**Comment by revam** [comment]
I was thinking of adding some "fancy" series guessing (spoiler; ~~it's regular expressions~~) while I was at it, and maybe some "fancy" episode number guessing too (~~using the same regexes~~).

**Comment by hidden4003** [comment]
It looks a lot simpler

**Comment by revam** [comment]
I'll do the series guessing and episode number guessing in another PR.

**Review by hidden4003 — APPROVED** [review]

**Pull request merged**